include/swift/SILOptimizer/PassManager/Passes.def

```diff
@@ -179,6 +179,12 @@ PASS(PredictableMemoryOptimizations, "predictable-memopt",
      "Predictable early memory optimizations")
 PASS(ReleaseDevirtualizer, "release-devirtualizer",
      "Devirtualize release-instructions")
+PASS(RetainSinking, "retain-sinking",
```

Comment:
Why do you need three passes?  Can't you implement a single ARC code motion with three phases?